astor/misc.py

```diff
@@ -170,6 +177,23 @@ def parsefile(fname):
     return ast.parse(fstr, filename=fname)
 
 
+def finfo(codeobj):
+    """Returns the file and line number of a code object.
+        If the code object has a __file__ attribute (e.g. if
+        it is a module), then the returned line number will
+        be 0
+
+    """
+    fname = getattr(codeobj, '__file__', None)
+    linenum = 0
+    if fname is None:
+        func_code = codeobj.__code__
+        fname = func_code.co_filename
+        linenum = func_code.co_firstlineno
```

Comment:
Perhaps we can leave this part to another helper or handle it in CodeToAst. It would be nice if the function do one thing.